purchases/src/main/kotlin/com/revenuecat/purchases/CoroutinesExtensions.kt

```diff
@@ -60,3 +61,45 @@ suspend fun Purchases.awaitOfferings(): Offerings {
         )
     }
 }
+
+data class PurchaseResult(
```

Comment:
Ideally we would use this as well for the non-coroutine version... But we would need to deprecate the existing ones I guess. I think it's ok to keep this to the coroutines version for now and move to this in a future PR.